saleor/dashboard/product/views.py

```diff
@@ -522,10 +519,54 @@ def attribute_delete(request, pk):
         return redirect('dashboard:product-attributes')
     return TemplateResponse(
         request,
-        'dashboard/product/product_attribute/modal/confirm_delete.html',
+        'dashboard/product/product_attribute/modal/'
+        'attribute_confirm_delete.html',
         {'attribute': attribute})
 
 
+@staff_member_required
+@permission_required('product.edit_properties')
+def attribute_choice_value_edit(request, attribute_pk, value_pk=None):
+    if value_pk:
+        value = get_object_or_404(AttributeChoiceValue, pk=value_pk)
+    else:
+        value = AttributeChoiceValue(attribute_id=attribute_pk)
+    form = forms.AttributeChoiceValueForm(
+        request.POST or None, instance=value)
+    if form.is_valid():
+        value = form.save()
+        msg = pgettext_lazy(
+            'Dashboard message', 'Updated attribute\'s value')\
+            if value.pk else pgettext_lazy(
+                'Dashboard message', 'Added attribute\'s value')
+        messages.success(request, msg)
+        return redirect('dashboard:product-attribute-detail', pk=attribute_pk)
+    ctx = {'attribute_pk': attribute_pk, 'value': value, 'form': form}
+    return TemplateResponse(
+        request,
+        'dashboard/product/product_attribute/values_add_edit.html',
+        ctx)
+
+
+@staff_member_required
+@permission_required('product.edit_properties')
+def attribute_choice_value_delete(request, attribute_pk, value_pk):
+    value = get_object_or_404(AttributeChoiceValue, pk=value_pk)
+    if request.method == 'POST':
+        value.delete()
+        messages.success(
+            request,
+            pgettext_lazy(
+                'Dashboard message',
+                'Removed attribute choice value %s') % (value.name,))
```

Comment:
This message is inconsistent with messages in the "edit" view. I think better would be: "Removed attribute's value: %s".